source/common/buffer/buffer_impl.cc

```diff
@@ -588,5 +588,42 @@ std::vector<Slice::SliceRepresentation> OwnedImpl::describeSlicesForTest() const
   return slices;
 }
 
+BufferMemoryAccountSharedPtr
+BufferMemoryAccountImpl::createAccount(WatermarkFactory* factory,
+                                       Http::StreamResetHandler* reset_handler) {
+  auto account =
```

Comment:
Might be worth adding a comment why shared_ptr ctor vs make_shared: B/c ctor is private!